pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -186,39 +207,80 @@ func (s *akvKMProvider) GetCertificates(ctx context.Context) (map[keymanagementp
 		logger.GetLogger(ctx, logOpt).Debugf("fetching secret from key vault, certName %v, certVersion %v, vaultURI: %v", keyVaultCert.Name, keyVaultCert.Version, s.vaultURI)
 
 		startTime := time.Now()
-		secretResponse, err := s.secretKVClient.GetSecret(ctx, keyVaultCert.Name, keyVaultCert.Version)
-		if err != nil {
-			if isSecretDisabledError(err) {
-				// if secret is disabled, get the version of the certificate for status
-				certResponse, err := s.certificateKVClient.GetCertificate(ctx, keyVaultCert.Name, keyVaultCert.Version)
-				if err != nil {
-					return nil, nil, fmt.Errorf("failed to get certificate objectName:%s, objectVersion:%s, error: %w", keyVaultCert.Name, keyVaultCert.Version, err)
+		versionHistory := []VersionInfo{}
+
+		certVersionPager := s.certificateKVClient.NewListCertificateVersionsPager(keyVaultCert.Name, nil)
```

Comment:
@shizhMSFT also pointed out the latest version is v1.16. https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime. Do we know if the latest API /SDK have improved behaviors?

Reply:
After looking into the latest version of the runtime package I didn't find any options that can be passed to sort the results of the pager. But there were some comments that indicated that it might be available in future releases. There are other azure-sdk-for-go packages that are deprecated and should be updated as well. Here's a link to an [issue](https://github.com/ratify-project/ratify/issues/1977) tracking that work.